### ISSUE TITLE
Mend Renovate 6.X does not honor job queue

### ISSUE THREAD

**Issue by ArchdukeNavaron** [mend/renovate-ce-ee#385]
Hi,

we recently migrated to the latest on-premise version of Renovate.
When we updated to 6.X, Renovate stopped prioritizing user inputs over scheduled jobs.

Unfortunately debug logs don't contain any hints why Renovate skips jobs with higher priority.

For example. We wanted Renovate to rebase a MR in GitLab.
We therefore checked the rebase checkbox which triggered a webhook.

```log
 INFO: Received webhook event (repository=<redacted>)
       "event": "merge_request"
DEBUG: Processed GitHub webhook event
       "event": {
         "action": "enqueue",
         "repositories": ["<redacted>"],
         "reason": "rebase-request",
         "installationId": 183,
         "organizationName": "<redacted>",
         "type": "external"
       }
```

Noteworthy: The log says "GitHub", but we use GitLab and configured it correctly. (Maybe that's just a hard coded log message)
We can also see the job in the queue with a very high priority.

```log
[
  {
    "jobId": 47866,
    "repository": "<redacted>",
    "organizationName": "<redacted>",
    "reason": "rebase-request",
    "priority": 120,
    "addedAt": "2023-11-03 09:14:03"
  },
  ...
]
```

But after Renovate finished with its current job, it randomly chooses another job and ignores the `rebase-request` job.

Would be very nice if anyone could help us with this.

**Comment by Gabriel-Ladzaretti** [comment]
> Noteworthy: The log says "GitHub", but we use GitLab and configured it correctly. (Maybe that's just a hard coded log message)

It was indeed a typo and was already addressed in `v6.4.0`.

> But after Renovate finished with its current job, it randomly chooses another job and ignores the rebase-request job.

Here is the [job priory mapping](https://github.com/mend/renovate-ce-ee/blob/main/docs/api.md#status) , a job is dequeued based on its priority and its age.
therefore, if two jobs have the same priory the older one will be processed first.

Furthermore, you can see the actual head of the queue by querying the `/api/job/queue` endpoint.
https://github.com/mend/renovate-ce-ee/blob/main/docs/api.md#queue-status

The `api/status` api can be useful as well.
https://github.com/mend/renovate-ce-ee/blob/main/docs/api.md#status

**Comment by Gabriel-Ladzaretti** [comment]
A job is dequeued not solely based on priority, it depends on whether the job was previously dequeued and failed. if it has failed, it will have a cooldown period before it is dequeued again, this prevents the worker from constantly attempting to scan problematic jobs. In other words, this prevents job starvation.

additionally, if a job fails three times in a row, there is an even longer cooldown period. to determine the current state of the job, you can query the job api and paste the result here. alternatively, you can use the bundled sqlite3 cli[^1] tool to interact directly with the database and retrieve the data as needed.

[^1]: https://github.com/mend/renovate-ce-ee/blob/main/docs/configure-renovate-ce-github.md#configure-mend-renovate-ce
